src/material/legacy-dialog/dialog.ts

```diff
@@ -0,0 +1,293 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Overlay, OverlayContainer, ScrollStrategy} from '@angular/cdk/overlay';
+import {ComponentType} from '@angular/cdk/portal';
+import {Location} from '@angular/common';
+import {
+  Inject,
+  Injectable,
+  InjectionToken,
+  Injector,
+  OnDestroy,
+  Optional,
+  SkipSelf,
+  TemplateRef,
+  Type,
+} from '@angular/core';
+import {defer, Observable, Subject} from 'rxjs';
+import {startWith} from 'rxjs/operators';
+import {MatLegacyDialogConfig} from './dialog-config';
+import {MatLegacyDialogContainer, _MatLegacyDialogContainerBase} from './dialog-container';
+import {MatLegacyDialogRef} from './dialog-ref';
+import {ANIMATION_MODULE_TYPE} from '@angular/platform-browser/animations';
+import {Dialog, DialogConfig} from '@angular/cdk/dialog';
+
+/** Injection token that can be used to access the data that was passed in to a dialog. */
+export const MAT_LEGACY_DIALOG_DATA = new InjectionToken<any>('MatDialogData');
+
+/** Injection token that can be used to specify default dialog options. */
+export const MAT_LEGACY_DIALOG_DEFAULT_OPTIONS = new InjectionToken<MatLegacyDialogConfig>(
+  'mat-dialog-default-options',
+);
+
+/** Injection token that determines the scroll handling while the dialog is open. */
+export const MAT_LEGACY_DIALOG_SCROLL_STRATEGY = new InjectionToken<() => ScrollStrategy>(
+  'mat-dialog-scroll-strategy',
+);
+
+/** @docs-private */
+export function MAT_LEGACY_DIALOG_SCROLL_STRATEGY_FACTORY(overlay: Overlay): () => ScrollStrategy {
+  return () => overlay.scrollStrategies.block();
+}
+
+/** @docs-private */
+export function MAT_LEGACY_DIALOG_SCROLL_STRATEGY_PROVIDER_FACTORY(
+  overlay: Overlay,
+): () => ScrollStrategy {
+  return () => overlay.scrollStrategies.block();
+}
+
+/** @docs-private */
+export const MAT_LEGACY_DIALOG_SCROLL_STRATEGY_PROVIDER = {
+  provide: MAT_LEGACY_DIALOG_SCROLL_STRATEGY,
+  deps: [Overlay],
+  useFactory: MAT_LEGACY_DIALOG_SCROLL_STRATEGY_PROVIDER_FACTORY,
+};
+
+// Counter for unique dialog ids.
+let uniqueId = 0;
+
+/**
+ * Base class for dialog services. The base dialog service allows
+ * for arbitrary dialog refs and dialog container components.
+ */
+@Injectable()
+export abstract class _MatLegacyDialogBase<C extends _MatLegacyDialogContainerBase>
```

Comment:
We should move the base class into the non-legacy version. That's what I've been doing in the other components since it'll be weird if the new components are importing legacy code.

Reply:
I don't see it as a big deal if the non-legacy one depends on the legacy one for now, obviously at some point before we remove the legacy one it will have to be moved though. I'll take a stab at just moving it as part of this PR to be consistent with what you've already done

Reply:
+1 I think we should move the base classes now (I had to do the same for tooltip)